feat: enable label enrichment for Metric and TracePipeline (#2195)
kyma-project/telemetry-manager: docs/user/gateways.md

  - Deployment/DaemonSet/StatefulSet/Job name
  - Namespace
  - Cluster name

> - `k8s.pod.label.<label_key>` attributes: In addition to the predefined enrichments, the Telemetry gateways support user-defined enrichments of telemetry data based on Pod labels. Users can configure specific label keys or label key prefixes to include in the enrichment process. This allows for the capture of custom application metadata that may be relevant for filtering, grouping, or correlation purposes. All matching Pod labels are added to the telemetry data as resource attributes, using original label key format `k8s.pod.label.<label_key>`.

Reviewer: ```suggestion
- `k8s.pod.label.<label_key>` attributes: In addition to the predefined enrichments, the Telemetry gateways support user-defined enrichments of telemetry data based on Pod labels. Users can configure specific label keys or label key prefixes to include in the enrichment process. This allows for the capture of custom application metadata that may be relevant for filtering, grouping, or correlation purposes. All matching Pod labels are added to the telemetry data as resource attributes, using the original label key format `k8s.pod.label.<label_key>`.
```

What does "original" mean here?

> The following example configuration shows how to enrich telemetry data with Pod labels that match the specified keys or key prefixes:

Reviewer 2: The blocks need to be intended to keep the list format

>    The following example configuration shows how to enrich telemetry data with Pod labels that match the specified keys or key prefixes:

Reviewer: I suggest we combine this sentence with the one from line 86ff.

```suggestion
   The following example configuration enriches the telemetry data with Pod labels that match the specified keys or key prefixes:
   - `k8s.pod.label.app.kubernetes.io/name`: The value of the exact label key `app.kubernetes.io/name` from the Pod.
   - `k8s.pod.label.app.kubernetes.io.*`: All labels that start with the prefix `app.kubernetes.io` from the Pod, where `*` is replaced by the actual label key.
```

```yaml
apiVersion: operator.kyma-project.io/v1alpha1
kind: Telemetry
metadata:
  name: default
  namespace: kyma-system
spec:
  enrichments:
    extractPodLabels:

>       - key: my.pod.label.exact.key

Author: The API is still experimental and stripped out, maybe we can link it after we remove the experimental flag.

>       - key: my.pod.label.exact.key
>       - keyPrefix: my.pod.label.prefix

Reviewer 2: ```suggestion
      - key: "<myExactLabelKey>" # for example "app.kubernetes.io/name"
      - keyPrefix: "<myLabelPrefix>" # for example "app.kubernetes.io"
```

>        - key: "<myExactLabelKey>" # for example "app.kubernetes.io/name"

Reviewer: ```suggestion
       - key: "<myExactLabelKey>" # for example, "app.kubernetes.io/name"
```

   ```

>    The above configuration enriches the telemetry data with the following resource attributes:

Reviewer: Added to line 72ff

```suggestion
```

>    `k8s.pod.label.app.kubernetes.io/name`: The value of the exact label key `app.kubernetes.io/name` from the Pod.

Reviewer: Added to line 72ff

```suggestion
```

>    `k8s.pod.label.app.kubernetes.io.*`: All labels that start with the prefix `app.kubernetes.io` from the Pod, where `*` is replaced by the actual label key.

Reviewer: Added to line 72ff

```suggestion
```
